gaphor/i18n.py

```diff
@@ -15,18 +17,47 @@
 
 log = logging.getLogger(__name__)
 
-if sys.platform == "win32" and os.getenv("LANG") is None:
-    language, _ = locale.getlocale()
-    if language:
-        os.environ["LANG"] = language
+
+def _get_os_language() -> str:
+    """Get the default language in Windows or macOS.
+
+    Inspired by https://github.com/nicotine-plus/nicotine-plus
+    locale.getlocale() fails to get the correct language if the region
+    is set different from the language.
+    """
+    if sys.platform == "win32":
+        import ctypes
+
+        windll = ctypes.windll.kernel32
+        if language := locale.windows_locale.get(windll.GetUserDefaultUILanguage()):
+            return language
+    elif sys.platform == "darwin":
+        import subprocess
+
+        try:
+            lang_out = subprocess.check_output(
+                ("defaults", "read", "-g", "AppleLanguages")
+            )
+            if languages := lang_out.decode("utf-8").strip('()\n" ').split(","):
+                return languages[0][:2]
```

Comment:
This output works with the LANGUAGE environment variable, but I don't remember if LANG supports it. Might be worth confirming this.

Reply:
I think LANGUAGE allows for a list of languages, but I expect here we are just getting the first one. Maybe @amolenaar would be able to help test on macOS that this works.

Reply:
Maybe https://stackoverflow.com/questions/17679651/how-to-figure-out-what-the-current-system-preferred-language-is? It isn't spawning any processes at least.

Reply:
Nice one, I'll update the PR!